lib/matplotlib/axes/_axes.py

```diff
@@ -3978,7 +3978,7 @@ def dopatch(xs, ys, **kwargs):
             # maybe draw the fliers
             if showfliers:
                 # fliers coords
-                flier_x = np.ones(len(stats['fliers'])) * pos
+                flier_x = np.full(len(stats['fliers']), pos, dtype='float64')
```

Comment:
I think dtype=float is generally preferrable, unless you have a good reason to force float64 even on 32bit architectures (which at least we didn't do before).

Reply:
actually, no, `dtype=float` is not preferable as it is referencing python floats. `dtype='float64'` or `dtype=np.float_` (if you don't want to specify the width) is generally considered preferable. Also note `dtype=np.int_` and `dtype=np.bool_` all exists for similar reasons.

Reply:
Eh... the link you quote literally says "Whenever a data type is required, as an argument, the standard Python types are recognized as well" and I don't think (?) there's a single quote in this statement saying that the dtypes are preferable in this use-case.

Reply:
My point was, and still is, that the fact that python datatypes are accepted is merely a convenience, and is not declared as "generally preferable". I have never seen anyone before state that the python datatypes are "generally preferable", and instead, I have seen people state the opposite because "explicit is better than implicit". I think what the submitter has is perfectly fine. And if the submitter had it as `dtype=float` in the original submission, I probably wouldn't have said anything anyway, either. I am only opposing the assertion that your suggestion was "generally preferable".

Reply:
To address the initial comment,
```py
>>> import numpy as np
>>> np.dtype(float)
dtype('float64')
>>> np.dtype(np.float64)
dtype('float64')
>>> np.dtype(np.float_)
dtype('float64')
```
so the actual dtype when providing `dtype=float` will be `float64`. That will be true for both 32 bit and 64 bit systems (checked in a 32 bit Debian docker container), unlike `np.dtype(int)` which can be `int32` or `int64` depending on the system.

I also think it's better to say `float64` explicitly even if it's 3 char longer, because that's  the actual numpy `dtype` we get, and it removes the need for readers to think about the sizeof float on 32/64 bit architecture and about possible implications of that.

Reply:
FWIW, `np.float64` is a slightly better spelling than `'float64'`